src/overlays/actors/ovl_En_Ru1/z_en_ru1.c

```diff
@@ -19,417 +135,2276 @@ const ActorInit En_Ru1_InitVars = {
+                gSaveContext.inf_table[20] |= 1;
+                return;
+            }
+        }
+    }
+    gSaveContext.inf_table[20] &= ~0x1;
+}
+
+s32 func_80AEE394(EnRu1* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    CollisionContext* colCtx;
+    DynaPolyActor* dynaActor;
+    s32 floorPolySource;
+
+    if ((this->actor.bgCheckFlags & 1) && (this->actor.floorPolySource != 0x32)) {
+        colCtx = &globalCtx->colCtx;
+        floorPolySource =
+            this->actor.floorPolySource; // necessary match, can't move this out of this block unfortunately
+        dynaActor = DynaPolyInfo_GetActor(colCtx, floorPolySource);
+        if ((dynaActor != NULL) && (dynaActor->actor.id == ACTOR_BG_BDAN_OBJECTS) && (dynaActor->actor.params == 0) &&
+            (!func_8008E988(globalCtx)) && (globalCtx->msgCtx.unk_E300 == 0)) {
+            func_80AEE02C(this);
+            globalCtx->csCtx.segment = &D_80AF10A4;
+            gSaveContext.cutscene_trigger = 1;
+            this->action = 36;
+            this->drawConfig = 0;
+            this->unk_28C = dynaActor;
+            this->actor.shape.unk_14 = 0;
+            return 1;
+        }
+    }
+    return 0;
+}
+
+void func_80AEE488(EnRu1* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->actor;
+    s8 curRoomNum;
+
+    if (func_8002F410(thisx, globalCtx)) {
+        curRoomNum = globalCtx->roomCtx.curRoom.num;
+        this->roomNum3 = curRoomNum;
+        this->action = 31;
+        func_80AED520(this, globalCtx);
+    } else if ((!func_80AEE394(this, globalCtx)) && (!(thisx->bgCheckFlags & 1))) {
+        thisx->minVelocityY = -((kREG(24) * 0.01f) + 6.8f);
+        thisx->gravity = -((kREG(23) * 0.01f) + 1.3f);
+        this->action = 28;
+    }
+}
+
+void func_80AEE568(EnRu1* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->actor;
+
+    if (!func_80AEE394(this, globalCtx)) {
+        if ((thisx->bgCheckFlags & 1) && (thisx->speedXZ == 0.0f) && (thisx->minVelocityY == 0.0f)) {
+            func_80AEE02C(this);
+            func_8002F580(this, globalCtx);
+            this->action = 27;
+            func_80AEADD8(this);
+        } else if (thisx->unk_84 > 0.0f) {
+            this->action = 29;
+            this->unk_350 = 0;
+        }
+    }
+}
+
+void func_80AEE628(EnRu1* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    s8 curRoomNum = globalCtx->roomCtx.curRoom.num;
+
+    if (func_80AEAF38(globalCtx)) {
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_06006B9C, 1.0f, 0, SkelAnime_GetFrameCount(&D_06006B9C.genericHeader),
+                             0, -8.0f);
+        gSaveContext.inf_table[20] |= 0x10;
+        this->action = 31;
+    }
+    this->roomNum3 = curRoomNum;
+}
+
+s32 func_80AEE6D0(EnRu1* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s8 curRoomNum = globalCtx->roomCtx.curRoom.num;
+
+    if ((!(gSaveContext.inf_table[20] & 0x10)) && (func_80AEB124(globalCtx) != 0)) {
+        if (func_8008E988(globalCtx) == 0) {
```

Comment:
probably combine

Reply:
These can't be combined because of the line after the condition: `this->roomNum3 = curRoomNum;`